python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -840,7 +890,11 @@ def reconcile(
     instance: "DagsterInstance",
     cursor: AssetReconciliationCursor,
     run_tags: Optional[Mapping[str, str]],
-):
+) -> Tuple[
+    Sequence[RunRequest],
+    AssetReconciliationCursor,
+    Mapping[AssetKeyPartitionKey, AbstractSet[AutoMaterializeReason]],
```

Comment:
Can we compress these by PartitionsSubset or something?

Reply:
we could, although I worry that if we over-specify our Materialize/SkipReasons (e.g. "We're skipping partition 2020-01-01 because upstream asset foo is missing partition 2020-01-01"), then we lose the ability to create meaningfully large groups here.

Maybe it's a reasonable outcome if we only keep that high-fidelity information for non-partitioned assets?

Reply:
Maybe we could record what asset we're waiting for but not what partition?